redfish-core/lib/systems.hpp

```diff
@@ -51,6 +55,56 @@
 namespace redfish
 {
 
+/* @brief validate systemName and get system index from systemName
+ *
+ * @param[in] systemName     requested system name
+ *
+ * @return int    -1 on failure, else validatet index
+ * */
+inline int getSystemIndex(const std::string_view systemName)
+{
+#ifndef BMCWEB_ENABLE_MULTI_COMPUTERSYSTEM
+    // on single host we only want system0 / system to be valid
+    if (systemName == "system0" || systemName == "system")
+    {
+        return 0;
+    }
+    return -1;
+#else
+
+    static constexpr std::string_view systemToken = "system";
+    auto location = systemName.rfind(systemToken);
+
+    if (location == std::string::npos)
+    {
+        BMCWEB_LOG_DEBUG("invalid host number");
+        return -1;
+    }
+
+    // TODO  12/05/23-14:37 olekatpyle: only works for single digit number
```

Comment:
TODO comments can be removed, implementation looks like it supports multi-digit indices.

Reply:
nvm i was tripping, you are right :D